extensions/interactions/ImageClickInput/validator.js

```diff
@@ -13,106 +13,122 @@
-        });
-      }
+        var areAnyRegionStringsEmpty = false;
+        var areAnyRegionStringsDuplicated = false;
+        var seenRegionStrings = [];
+        if (argsValue.labeledRegions.length == 0) {
+          warningsList.push({
+            type: WARNING_TYPES.ERROR,
+            message: 'Please specify at least one image region to click on.'
+          });
+        }
+
+        for (var i = 0; i < argsValue.labeledRegions.length; i++) {
+          var regionLabel = (
+            argsValue.labeledRegions[i].label);
 
-      for (var i = 0;
-           i < customizationArgs.imageAndRegions.value.labeledRegions.length;
-           i++) {
-        var regionLabel = (
-          customizationArgs.imageAndRegions.value.labeledRegions[i].label);
+          var ALPHANUMERIC_REGEX = /^[A-Za-z0-9]+$/;
+          if (regionLabel.trim().length === 0) {
+            areAnyRegionStringsEmpty = true;
+          } else if (!ALPHANUMERIC_REGEX.test(regionLabel)) {
+            warningsList.push({
+              type: WARNING_TYPES.CRITICAL,
+              message: (
+                'The image region strings should consist of characters from ' +
+                '[A-Za-z0-9].')
+            });
+          } else if (seenRegionStrings.indexOf(regionLabel) !== -1) {
+            areAnyRegionStringsDuplicated = true;
+          } else {
+            seenRegionStrings.push(regionLabel);
+          }
+        }
 
-        var ALPHANUMERIC_REGEX = /^[A-Za-z0-9]+$/;
-        if (regionLabel.trim().length === 0) {
-          areAnyRegionStringsEmpty = true;
-        } else if (!ALPHANUMERIC_REGEX.test(regionLabel)) {
+        if (areAnyRegionStringsEmpty) {
           warningsList.push({
             type: WARNING_TYPES.CRITICAL,
-            message: (
-              'The image region strings should consist of characters from ' +
-              '[A-Za-z0-9].')
+            message: 'Please ensure the image region strings are nonempty.'
           });
-        } else if (seenRegionStrings.indexOf(regionLabel) !== -1) {
-          areAnyRegionStringsDuplicated = true;
-        } else {
-          seenRegionStrings.push(regionLabel);
         }
-      }
+        if (areAnyRegionStringsDuplicated) {
+          warningsList.push({
+            type: WARNING_TYPES.CRITICAL,
+            message: 'Please ensure the image region strings are unique.'
+          });
+        }
+        return warningsList;
+      },
+      getAnswerGroupWarnings: function(
+          customizationArgs, answerGroups, stateName) {
+        var warningsList =
+          baseInteractionValidationService.getAnswerGroupWarnings(
+            answerGroups, stateName);
 
-      if (areAnyRegionStringsEmpty) {
-        warningsList.push({
-          type: WARNING_TYPES.CRITICAL,
-          message: 'Please ensure the image region strings are nonempty.'
+        var argsValue = customizationArgs.imageAndRegions.value;
```

Comment:
imageAndRegionsValue, perhaps. "argsValue" sounds like it refers to all the args.